packages/next/src/build/next-config-ts/transpile-config.ts

```diff
@@ -0,0 +1,83 @@
+import type { CompilerOptions } from 'typescript'
+import type { Options as SWCOptions } from '@swc/core'
+import { readFile } from 'node:fs/promises'
+import { join } from 'node:path'
+import { deregisterHook, registerHook, requireFromString } from './require-hook'
+import { transform } from '../swc'
+
+function resolveSWCOptions(
+  cwd: string,
+  compilerOptions: CompilerOptions
+): SWCOptions {
+  const resolvedBaseUrl = join(cwd, compilerOptions.baseUrl ?? '.')
+  return {
+    jsc: {
+      target: 'es5',
+      parser: {
+        syntax: 'typescript',
+      },
+      paths: compilerOptions.paths,
+      baseUrl: resolvedBaseUrl,
+    },
+    module: {
+      type: 'commonjs',
+    },
+    isModule: 'unknown',
+  } satisfies SWCOptions
+}
+
+// since tsconfig allows JS comments (jsonc), strip them for JSON.parse
+function stripComments(str: string) {
+  return str.replace(/\\"|"(?:\\"|[^"])*"|(\/\/.*|\/\*[\s\S]*?\*\/)/g, (m, g) =>
+    g ? '' : m
+  )
+}
+
+async function lazilyGetTSConfig(cwd: string) {
+  let tsConfig: { compilerOptions: CompilerOptions }
+  try {
+    tsConfig = JSON.parse(
+      stripComments(await readFile(join(cwd, 'tsconfig.json'), 'utf8'))
+    )
```

Comment:
We can re-use our `parseJsonFile` function which handles comments already here https://github.com/vercel/next.js/blob/4e84f696f057377893fc34680c3dc535a0e77a3b/packages/next/src/build/load-jsconfig.ts#L12

Reply:
Thank you!!